src/mscorlib/src/System/TimeZoneInfo.Unix.cs

```diff
@@ -381,26 +381,34 @@ private static bool TryLoadTzFile(string tzFilePath, ref byte[] rawData, ref str
         /// </summary>
         private static string FindTimeZoneIdUsingReadLink(string tzFilePath)
         {
-            string id = null;
-
-            StringBuilder symlinkPathBuilder = StringBuilderCache.Acquire(Path.MaxPath);
-            bool result = Interop.GlobalizationInterop.ReadLink(tzFilePath, symlinkPathBuilder, (uint)symlinkPathBuilder.Capacity);
-            if (result)
+            int capacity = 4096;
+            do
```

Comment:
Is it worth actually checking the size first (lstat)? What is the "normal" expected case for the first attempt otherwise?

Reply:
I don't think that's useful. readlink will inform when the file doesn't exist. capacity refers to the file name, not the file content.